src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.cs

```diff
@@ -187,6 +187,30 @@ protected internal override void OpenPin(int pinNumber)
                     throw ExceptionHelper.GetIOException(ExceptionResource.OpenPinError, Marshal.GetLastWin32Error());
                 }
 
+                int mode = Interop.libgpiod.gpiod_line_direction(pinHandle);
+                if (mode == 1)
+                {
+                    pinHandle.PinMode = PinMode.Input;
+                }
+                else if (mode == 2)
+                {
+                    pinHandle.PinMode = PinMode.Output;
+                }
+
+                if (s_isLibgpiodVersion1_5orHigher && pinHandle.PinMode == PinMode.Input)
+                {
+                    int bias = Interop.libgpiod.gpiod_line_bias(pinHandle);
+                    if (bias == (int)RequestFlag.GPIOD_LINE_REQUEST_FLAG_BIAS_PULL_DOWN)
+                    {
+                        pinHandle.PinMode = PinMode.InputPullDown;
+                    }
+
+                    if (bias == (int)RequestFlag.GPIOD_LINE_REQUEST_FLAG_BIAS_PULL_UP)
+                    {
+                        pinHandle.PinMode = PinMode.InputPullUp;
+                    }
```

Comment:
what other options there are? Should we throw or do something if it's out of the known range?

Reply:
Currently, only these values are known. See also above